### DISCUSSION TITLE
The guide to Git I never had. | Glasskube.dev

### DISCUSSION THREAD

**Discussion by giscus** [glasskube/glasskube#705] [Guides]
# The guide to Git I never had. | Glasskube.dev

Learn how to use git with commands for commiting, branching, rebasing and squashing

https://glasskube.dev/guides/git/

**Comment by keithpenney** [comment]
Thanks for the great discussion!  I consider `git reset` to be the proper antonym to `git add` since it simply removes a file from the staging area, rather than queuing it up for removal in the next commit.  `git rm` is great for those (embarrassingly frequent) times you accidentally check in some temp file or build artifact (pro tip: squash the history), or when rebasing deprecates an entire file.  [This SO post](https://stackoverflow.com/questions/33004187/what-is-the-practical-difference-between-git-rm-cached-git-reset-and) is helpful too.
